### PR TITLE
Added Docker and docker compose files

### PR DESCRIPTION
**Pull request by ansultan1**
i have containerized the app with docker and docker compose. Now bringing up this application stack is as easy as running the docker compose up command.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DB_HOST="mongodb"
+DB_PORT="27017"
+DB_NAME="task_schedular"
+SERVER_HOST="0.0.0.0"
+SERVER_PORT="8080"
```
Comment (OvaisTariq95) [inline .env]: Add newline at the end of file

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.21-rc-alpine
+WORKDIR /app
+COPY . .
+RUN go mod tidy
+EXPOSE 8080
+CMD ["go", "run", "cmd/main.go"]
```
Comment (OvaisTariq95) [inline Dockerfile]: Add newline at the end of file

File: docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      - mongodb:/var/lib/mongodb/data
+    networks:
+      - go-network
+
+volumes:
+  mongodb:
+
+networks:
+  go-network:
+    driver: bridge
```
Comment (OvaisTariq95) [inline docker-compose.yaml]: Add newline at the end of file

**Review by OvaisTariq95 — CHANGES_REQUESTED** [review]
Good work!

File: docker-compose.yaml (hunk 3 above)
Comment (ansultan1) [reply docker-compose.yaml]: resolved

**Pull request merged**